### DISCUSSION TITLE
`scan`ing or similar over a list of functions

### DISCUSSION THREAD

**Discussion by pabloduque0** [jax-ml/jax#12130] [Q&A] [answered]
Lets assume we have a list of functions (like data augmentation functions for example) in which all transformations are applied to the output of the previous function in a sequential way. Something like:

```
images = jnp.ones((25, 100, 100, 3))
augmentations = [fn1, fn2, fn3]

for aug in augmentations:
    images = aug(images)
```

`scan` seemed like a great choice since I can rely on the `carry` but after a couple attempts I think I cannot scan over a list of functions. `vmap` does not necessarily fit in here either and I think we cannot `vmap` over such list anyways.

I was wondering if there is a way to speed up this beyond `jit`ing the function where this logic happens. Maybe there is actually not greater benefit from trying something else and `jit` could be as good as we are going to get.

Thank you very much in advance!

**Comment by jakevdp** [comment, marked as answer]
It is only possible to scan over a list of values, not over a list of functions. The best way to achieve what you have in mind is likely with a `for` loop, as you used in your example above.

**Reply by pharringtonp19** [reply]
@pabloduque0 I had a [similar question](https://github.com/google-research/dex-lang/issues/962) that I asked to the folks leading the [dex-lang](https://github.com/google-research/dex-lang) project. You might find the comments there helpful as well.

**Reply by pabloduque0** [reply]
Thank you both for the answers, will take a look at that!
